components/builder-dbcache/src/data_store.rs

```diff
@@ -88,6 +88,13 @@ pub trait BasicSet: Bucket {
     /// Type of objects stored inside this data set.
     type Record: Persistable;
 
+    /// Delete a record from the data set with the given ID.
```

Comment:
I think you're right that most things we want to "taint" or "deprecate" instead of delete which is probably why it took until now to add a function for deleting something. This permanent delete is only used in the administration stuff right now becuase it's what we actually need, but in the depot for example we won't use this at all most likely